ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp

```diff
@@ -0,0 +1,282 @@
+  m_planeGraphic->attributes()->insertAttribute("PITCH", 0.0);
+  m_planeGraphic->attributes()->insertAttribute("HEADING", 45.0);
+
+  //Add the plane graphic to the scene.
+  m_sceneView->graphicsOverlays()->at(0)->graphics()->append(m_planeGraphic);
+
+
+  /* Orbiting Camera setup */
+  // create the camera controller to follow the plane
+  m_orbitCam = new OrbitGeoElementCameraController(m_planeGraphic, 50, this);
+
+  // restrict the camera's heading to stay behind the plane
+  // Note : Since this is exposed and hooked up as a property, the heading slider in the UI will automatically adjust its bounds, try changing these values.
+  m_orbitCam->setMinCameraHeadingOffset(-45);
+  m_orbitCam->setMaxCameraHeadingOffset(45);
+  emit cameraHeadingBoundsChanged();
+
+  // restrict the camera's pitch so it doesn't go completely vertical or collide with the ground
+  m_orbitCam->setMinCameraPitchOffset(10);
+  m_orbitCam->setMaxCameraPitchOffset(100);
+
+  // restrict the camera to stay between 10 and 1000 meters from the plane
+  m_orbitCam->setMinCameraDistance(10);
+  m_orbitCam->setMaxCameraDistance(100);
+
+  // position the plane a third from the bottom of the screen
+  m_orbitCam->setTargetVerticalScreenFactor(0.33f);
+  // don't pitch the camera when the plane pitches
+  m_orbitCam->setAutoPitchEnabled(false);
+
+  // Forward the camera heading changed signal on to the QML property signal.
+  // This allows the "camera heading" slider to react to changing the camera heading via mouse motion
+  // See "onMoved" signal handler in OrbitCameraAroundObject.qml
+  connect(m_orbitCam, &OrbitGeoElementCameraController::cameraHeadingOffsetChanged, this, &OrbitCameraAroundObject::cameraHeadingChanged);
+
+  //Apply our new orbiting camera to the scene view.
+  m_sceneView->setCameraController(m_orbitCam);
+
+  emit sceneViewChanged();
+}
+
+void OrbitCameraAroundObject::cockpitView()
+{
+  /*
+   * Animates the camera to a cockpit view. The camera's target is offset to the cockpit (instead of the plane's
+   * center). The camera is moved onto the target position to create a swivelling camera effect. Auto pitch is
+   * enabled so the camera pitches when the plane pitches.
+   */
+
+  //No zooming in/out by default in cockpit view
+  setCamDistanceInteractionAllowed(false);
+
+  //allow the camera to get closer to the target
+  m_orbitCam->setMinCameraDistance(0);
+
+  // animate the camera target to the cockpit instead of the center of the plane
+  m_orbitCam->setTargetOffsets(0, -2, 1.1, 1);
+
+  //The animation may rotate us over the set camera bounds based on the plane pitch, so unlock them.
+  m_orbitCam->setMinCameraPitchOffset(-180.0);
+  m_orbitCam->setMaxCameraPitchOffset(180.0);
+
+  //Camera move-to-cockpit callback.
+  connect(m_orbitCam, &OrbitGeoElementCameraController::moveCameraCompleted, this,
+          [this](QUuid, bool succeeded)
+  {
+    if(succeeded)
```

Comment:
our coding standard calls for a space between the if and the parenthesis - `if (succeeded)`